### PR TITLE
Automation sprint zero

### PR DESCRIPTION
**Pull request by luisczi**
### Summary:
- **What:** `<brief description>`
- **Ticket:** [sc<fill_in_issue_number>](https://app.shortcut.com/genepi/story/<fill_in_issue_number>)
- **Env:** `<rdev link>`

### Checklist:
- [ ] I merged latest `<base branch>`
- [ ] I manually verified the change
- [ ] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "devDependencies": {
+    "@playwright/test": "^1.23.3",
+    "babel-preset-env": "^1.7.0",
+    "cypress-wait-until": "^1.7.2",
```
Comment (phoenixAja) [inline package.json:5]: i don't think we need this cypress dependency any more

File: .gitignore [hunk 2]
```diff
@@ -309,3 +309,4 @@ static
 
 # Next.js
 .next
+/src/frontend/playwright/storage/
```
Comment (phoenixAja) [inline .gitignore:312]: you already have playright storage in the frontend gitignore file, do you need it here too?

File: src/frontend/README-E2E.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# cypress
+## installation
+Run `npm i` to install the Cypress from the package.json.
+
+## running tests
+To run Cypress tests you will need to specify the config file. We have created NPM scripts to make like a bit easier:
+
+`npm run cy:staging:open` - runs Cypress tests in UI runner and browser. You will be able to follow the progress and results of the test in the Cypress UI and the target application in the browser
+`npm run cy:staging:run` - this will run Cypress tests in the terminal without the UI and browser. The results will be printed in the Mocha format.
```
Comment (phoenixAja) [inline src/frontend/README-E2E.md:25]: can remove this

File: src/frontend/README-E2E.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+NOTES: 
+You need to be in the `src/frontend` to run these tests
```
Comment (phoenixAja) [inline src/frontend/README-E2E.md:28]: would also be nice to include the make commands for running (`make frontend-test`) and e2e when that's fixed

File: src/frontend/e2e/setup/global-setup.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+): Promise<void> {
+  const browser = await chromium.launch();
+  context = await browser.newContext();
+
+  const page = await browser.newPage();
+  await login(page, username, password);
+  // Set session storage in a new context
+  const sessionStorage = process.env.SESSION_STORAGE;
+  console.log("Reading session from storage");
+  console.log(sessionStorage);
```
Comment (phoenixAja) [inline src/frontend/e2e/setup/global-setup.ts:18]: remove the console.logs

File: src/frontend/e2e/tests/filters.spec.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import {test,expect} from '@playwright/test'
+import {filterGenomeRecovery,getSampleStatuses,filterLineage} from '../utils/archive/helpers'
+
+
+test.describe("Filter Test Suite",()=>{
+    test.beforeEach(async ({ page })=>{
+        await page.goto('https://staging.czgenepi.org/');     
+        await page.click('a[data-test-id="navbar-sign-in-link"]');
+        await page.type('#username','lbrambila@contractor.chanzuckerberg.com');
+        await page.type('#password','Br@mb1la');
```
Comment (phoenixAja) [inline src/frontend/e2e/tests/filters.spec.ts:10]: why not use the `login(page, testInfo);` like some of the other tests are doing? we should get the actual username and password from the enviornment, see [example here](https://github.com/chanzuckerberg/czgenepi/blob/94e0c592245f4f572fdddf70cb26f248db1aab23/src/frontend/tests/features/data.test.ts#L114)

File: src/frontend/e2e/tests/filters.spec.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        await filterGenomeRecovery('Complete',page);
+        const filterOn = await page.locator('div > .MuiChip-deletable');
+        await expect(filterOn).toHaveText('Complete');
+        await getSampleStatuses(page,'complete');
+    })
+    
+    test.skip("Genome Recovery filter by [failed] test",async ({ page })=>{
+        await expect(page).toHaveURL('https://staging.czgenepi.org/data/samples');
+        await filterGenomeRecovery('failed',page);
+        const filterOn = await page.locator('div > .MuiChip-deletable');
```
Comment (phoenixAja) [inline src/frontend/e2e/tests/filters.spec.ts:27]: we should have a test-id here that's more specific, see [here](https://github.com/chanzuckerberg/czgenepi/blob/94e0c592245f4f572fdddf70cb26f248db1aab23/src/frontend/tests/features/data.test.ts#L43) for examples

File: .gitignore (hunk 2 above)
Comment (luisczi) [reply .gitignore:312]: mmm actually these changes were already in, i think i might changed 2 files only

File: src/frontend/e2e/setup/local.config.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    trace: 'on-first-retry',
+    screenshot: "only-on-failure",
+    ignoreHTTPSErrors: true,
+    viewport: {width: 1280, height: 720}
+  },
+  projects: [
+    {
+        name: 'chromium',
+        use: {
+            ...devices['Desktop Chrome']
```
Comment (phoenixAja) [inline src/frontend/e2e/setup/local.config.ts:28]: would also like to check against safari, firefox, etc.

File: src/frontend/e2e/setup/staging.config.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  testDir: '../tests',
+  timeout: 30000,
+  expect: {
+    timeout: 30000,
+  },
+  reporter: [ ['html', {open: 'never'} ] ],
+  globalSetup: './global-setup',
+  use: {
+    baseURL: "https://staging.czgenepi.org",
+    //storageState: 'playwright/storage/state.json',
```
Comment (phoenixAja) [inline src/frontend/e2e/setup/staging.config.ts:17]: remove?

File: src/frontend/e2e/tests/filters.spec.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+test.describe("Filter Test Suite",()=>{
+    test.beforeEach(async ({ page })=>{
+        await page.goto('https://staging.czgenepi.org/');     
+        await page.click('a[data-test-id="navbar-sign-in-link"]');
+        await page.type('#username','lbrambila@contractor.chanzuckerberg.com');
+        await page.type('#password','Br@mb1la');
+        await page.click("main.login button[type='submit']");
+        await expect(page).toHaveURL('https://staging.czgenepi.org/data/samples');
+        await page.waitForURL("https://staging.czgenepi.org/data/samples");
```
Comment (phoenixAja) [inline src/frontend/e2e/tests/filters.spec.ts:13]: you should also be picking up this from an ENV variable (`ENV.API_URL`)

File: src/frontend/e2e/tests/home.spec.ts [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    "Careers":  "https://chanzuckerberg.com/careers/career-opportunities/?initiative=science",
+    "Learning Center":  "https://help.czgenepi.org/hc/en-us/categories/6217716150804-Genomic-Epidemiology-Learning-Center",
+    //"Privacy":  "/privacy",
+    //"Terms":  "/terms",
+    //"Contact Us":  "mailto:hello@czgenepi.org"
+}
+
+test.describe("Home page tests", () => {
+  test('Should verify home page', async ({ page }) => {
+    await page.goto('https://staging.czgenepi.org');
```
Comment (phoenixAja) [inline src/frontend/e2e/tests/home.spec.ts:15]: pick this up from env as well

**Review by phoenixAja — CHANGES_REQUESTED** [review]

**Comment by luisczi** [comment]
i need to upload newer changes and then reopen pull request

**Pull request closed without merging**